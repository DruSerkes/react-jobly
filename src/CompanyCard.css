.CompanyCard {
	margin: 1rem auto;
	background-color: white;
	width: 50%;
	box-shadow: 1px 2px 3px lightgray;
	padding: 1rem;
	border-radius: 1px;
	position: relative;
	opacity: 0.8;
	color: #2f2a2a;
}

.CompanyCard:hover {
	box-shadow: 1px 2px 3px lightgray, -1px -0px 3px lightgray;
	opacity: 1;
}

.CompanyCard h3 {
	display: flex;
	justify-content: space-between;
}

.CompanyCard span {
	text-align: left;
}

.CompanyCard p {
	text-align: left;
	width: 100%;
	margin: 1rem;
	margin: 10px 0;
}

.CompanyCard img {
	font-weight: 200;
}
