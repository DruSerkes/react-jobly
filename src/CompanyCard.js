import React from 'react';
import './CompanyCard.css';

const CompanyCard = ({ company }) => {
	return (
		<div className="CompanyCard">
			<h3>{company.name}</h3>
			<img src={company.logo_url} alt={company.name} />
			<p>{company.description}</p>
		</div>
	);
};

export default CompanyCard;
