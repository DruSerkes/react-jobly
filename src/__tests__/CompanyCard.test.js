import React from 'react';
import { render } from '@testing-library/react';
import CompanyCard from '../CompanyCard';

describe('CompanyCard tests', () => {
	const mockCompany = {
		name        : 'mock',
		description : 'this is a mock company',
		logo_url    : 'https://www.test.com/img.png'
	};

	it('renders without breaking', () => {
		render(<CompanyCard company={mockCompany} />);
	});

	it('matches snapshot', () => {
		const { asFragment } = render(<CompanyCard company={mockCompany} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('contains name and description', () => {
		const { getByText } = render(<CompanyCard company={mockCompany} />);
		const name = getByText(mockCompany.name);
		const description = getByText(mockCompany.description);
		expect(name).toBeInTheDocument();
		expect(description).toBeInTheDocument();
	});
});
