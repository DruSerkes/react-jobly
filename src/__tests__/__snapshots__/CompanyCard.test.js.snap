// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard"
  >
    <h3>
      <span>
        mock
      </span>
      <img
        alt="mock"
        src="https://www.test.com/img.png"
      />
    </h3>
    <p>
      this is a mock company
    </p>
  </div>
</DocumentFragment>
`;
